release/lib/update-dependencies.ts

```diff
@@ -47,30 +122,49 @@ const updateProperty = (property: string) => {
              * version is smaller. If this happens this means that something has gone terribly
              * wrong before or the user should make sure to rebase with the latest `master`.
              */
-            if (pkgVersion !== dependencyVersion) {
-                pkg.content[property][dependencyName] = dependencyVersion;
-                pkg.updated = true;
+            if (pkgVersion === dependencyVersion) {
+                return;
+            }
+
+            /*
+             * Ignore non-major optional dependency changes to avoid circular version updates.
+             * Otherwise these bump nearly all packages due to `hint` referencing configurations.
+             */
+            if (property === 'optionalDependencies' && !isMajorBump(pkgVersion, dependencyVersion)) {
+                return;
+            }
+
+            pkg.content[property][dependencyName] = dependencyVersion;
+            pkg.updated = true;
+
+            /*
+             * Update the version of this package (`patch`) if it has not been bumped yet
+             * (because it has changed versions in its dependencies).
+             */
+            if (pkg.oldVersion === pkg.content.version && pkg.updated) {
```

Comment:
You don't need `&& pkg.updated` because is going to be always true.

Reply:
Good point. I'm actually kind of surprised neither TypeScript nor ESLint flagged that 🤔